src/PullRequestReleaseNotes/Providers/GitHubPullRequestProvider.cs

```diff
@@ -40,28 +40,28 @@ public PullRequestDto Get(string commitMessage)
             var pullRequestId = ExtractPullRequestNumber(commitMessage);
             if (pullRequestId == null)
                 return null;
-            var pullRequest = _gitHubClient.PullRequest.Get(_programArgs.GitHubOwner, _programArgs.GitHubRepository, (int) pullRequestId).Result;
-            if (pullRequest == null)
-                return null;
-            if (_programArgs.VerboseOutput)
-                Console.WriteLine($"Found #{pullRequest.Number}: {pullRequest.Title}");
-            return GetPullRequestWithLabels(pullRequest);
+            return GetPullRequestWithLabels(pullRequestId.Value);
```

Comment:
> I think we can gain a good performance boost by using Octokit in an async way rather than synchronously. Do you want to help me look at introducing use of async in the PullRequestProviders?

Technically, async will slow this application down because the overhead of creating new threads. Async has two useful scenarios:
- For desktop application, we can use `async/await/BackGroundworker/Thread/Task` to delegate the work to a background thread; therefore the UI won't be blocked.
- For web server, we can `await` on IO (including networking) operations, so the thread can be used can handle another request instead of waiting for the result from IO. 

To make this application faster, probably just create multiple threads to perform the GitHub requests in parallel because the network is not fully utilised. To acheive this, we just need to change the `unreleasedCommits.Select(...` to `unreleasedCommits.AsParallel().Select(...`. A problem is I'm not sure is there a limit of concurrent requests from BitBucket/GitLab/TFS side, github seems to be fine. If it's a concern for BitBucket/GitLab/TFS, we can move the `GetPullRequests` to the `PullRequestProvider` so for github we can request in parallel while request sequentially for other providers.